common/src/main/java/gov/cms/ab2d/common/service/JobServiceImpl.java

```diff
@@ -185,16 +186,18 @@ public Resource getResourceForJob(String jobUuid, String fileName) throws Malfor
 
     @Override
     public void deleteFileForJob(File file, String jobUuid) {
-        boolean deleted = file.delete();
-        if (!deleted) {
-            log.error("Was not able to delete the file {}", file.getName());
-        }
-
         String fileName = file.getName();
         Job job = jobRepository.findByJobUuid(jobUuid);
         JobOutput jobOutput = jobOutputService.findByFilePathAndJob(fileName, job);
         jobOutput.setDownloaded(true);
         jobOutputService.updateJobOutput(jobOutput);
+        eventLogger.log(new FileEvent(
+                job == null || job.getUser() == null ? null : job.getUser().getUsername(),
+                jobUuid, file, FileEvent.FileStatus.DELETE));
+        boolean deleted = file.delete();
+        if (!deleted) {
+            log.error("Was not able to delete the file {}", file.getName());
+        }
```

Comment:
Did this need to be moved from the beginning of the method to the end?

Reply:
Since this block isn't throwing any exceptions, the move does not seem to have any effect.

Reply:
You need to create the event before the file is deleted (otherwise, you can't get the hash or file size) but still need to populate it with job information.